### PR TITLE
[#162311] New reconciliation status for "abandoned" orders

### PR DESCRIPTION
**Pull request by LeticiaErrandonea**
# Release Notes
NOTE: run the rake task on heroku after deploying

* Admins can now mark as "Unrecoverable" statements that are unreconciled.

# Screenshot
## Reconcile
image

## Mark as "Unrecoverable"
image

## Statement history
image

# Additional Context
## Pending work
- [ ] Specs

# Accessibility
- [ ] Did you scan for accessibility issues?
- [ ] Did you check our accessibility goal checklist?
- [ ] Did you add accessibility [specs](https://github.com/dequelabs/axe-core-gems/blob/develop/packages/axe-core-rspec/README.md)?

### CODE REVIEW

File: app/services/order_details/reconciler.rb [hunk 1]
```diff
@@ -12,8 +12,17 @@ class Reconciler
...
-    def initialize(order_detail_scope, params, reconciled_at, bulk_reconcile_note = nil, bulk_deposit_number = nil, bulk_reconcile_checkbox = nil)
+    def initialize(
+      order_detail_scope,
+      params,
+      reconciled_at,
+      bulk_reconcile_note = nil,
+      bulk_deposit_number = nil,
+      bulk_reconcile_checkbox = nil,
+      order_status = "reconciled"
+    )
```
Comment (giladshanan) [inline app/services/order_details/reconciler.rb:23]: I like this - it's much easier to read this way.  I think it should be indented to the right a bit though?

File: app/services/order_details/reconciler.rb [hunk 2]
```diff
@@ -44,12 +53,20 @@ def to_be_reconciled
...
-    def reconcile(order_detail, params)
-      order_detail.reconciled_at = @reconciled_at
-      order_detail.assign_attributes(allowed(params))
-      order_detail.reconciled_note = @bulk_reconcile_note if @bulk_reconcile_note.present?
-      order_detail.deposit_number = @bulk_deposit_number if @bulk_deposit_number.present?
-      order_detail.change_status!(OrderStatus.reconciled)
+    def update_status(order_detail, params)
+      if @order_status == "reconciled"
+        order_detail.reconciled_at = @reconciled_at
+        order_detail.assign_attributes(allowed(params))
```
Comment (giladshanan) [inline app/services/order_details/reconciler.rb:59]: nit pick - I think this line runs in both branches, so probably you could do this first before checking `@order_status`

File: app/views/facility_accounts_reconciliation/_action_row.html.haml [hunk 3]
```diff
@@ -1,27 +1,30 @@
 .row.table-actions.form-horizontal
   .span2.select_all_none= select_all_link
+  - if show_order_status
+    = select_tag :order_status, options_for_select([['Reconciled', 'reconciled'], ['Unrecoverable', 'unrecoverable']]), class: "js--orderStatusSelect"
```
Comment (giladshanan) [inline app/views/facility_accounts_reconciliation/_action_row.html.haml:4]: Double quotes please

**Review by giladshanan — APPROVED** [review]
A few tiny nitpicks, looks good

File: app/models/order_detail.rb [hunk 4]
```diff
@@ -752,7 +762,9 @@ def cancel_reservation(canceled_by, canceled_reason: nil, order_status: OrderSta
         change_status! order_status
       end
     else
+      return false if state.to_sym == :unreconciled
```
Comment (giladshanan) [inline app/models/order_detail.rb:765]: Is this intended to be `:unreconciled`? 
I wonder if this is part of what is making specs fail
```suggestion
      return false if state.to_sym == :unrecoverable
```

File: app/views/facility_accounts_reconciliation/_action_row.html.haml [hunk 5]
```diff
@@ -1,27 +1,30 @@
 .row.table-actions.form-horizontal
   .span2.select_all_none= select_all_link
+  - if show_order_status
```
Comment (giladshanan) [inline app/views/facility_accounts_reconciliation/_action_row.html.haml:3]: I wonder if we use this "action_row" partial in other places, like Move up, that could be causing spec failures.  
```suggestion
  - if local_assigns[:show_order_status]
```

**Review by giladshanan — COMMENTED** [review]
@LeticiaErrandonea looks like some specs are failing.

File: app/models/order_detail.rb [hunk 6]
```diff
@@ -752,7 +762,9 @@ def cancel_reservation(canceled_by, canceled_reason: nil, order_status: OrderSta
         change_status! order_status
       end
     else
+      return false if state.to_sym == :unrecoverable
```
Comment (giladshanan) [inline app/models/order_detail.rb:765]: Why is this here?

File: lib/tasks/demo_seed.rake [hunk 7]
```diff
@@ -15,6 +15,7 @@ namespace :demo do
     canceled   = OrderStatus.find_or_create_by!(name: "Canceled")
     complete   = OrderStatus.find_or_create_by!(name: "Complete")
     reconciled = OrderStatus.find_or_create_by!(name: "Reconciled")
+    abandoned  = OrderStatus.find_or_create_by!(name: "Unrecoverable")
```
Comment (giladshanan) [inline lib/tasks/demo_seed.rake:18]:
```suggestion
    unrecoverable  = OrderStatus.find_or_create_by!(name: "Unrecoverable")
```

File: spec/services/order_details/reconciler_spec.rb [hunk 8]
```diff
@@ -12,7 +12,7 @@
     end
   end
   let(:params) { order_details.each_with_object({}) { |od, h| h[od.id.to_s] = ActionController::Parameters.new(reconciled: "1") } }
-  let(:reconciler) { described_class.new(OrderDetail.all, params, Time.current) }
+  let(:reconciler) { described_class.new(OrderDetail.all, params, Time.current, "reconciled") }
```
Comment (giladshanan) [inline spec/services/order_details/reconciler_spec.rb:15]: Are you adding these just to be explicit, right?  The specs would pass without this?

**Review by giladshanan — COMMENTED** [review]
Left some questions

File: app/models/order_detail.rb (hunk 6 above)
Comment (LeticiaErrandonea) [reply app/models/order_detail.rb:765]: I don't know if we'll have this case, but it made sense to prevent a reservation from being canceled if the work had already been done

File: spec/services/order_details/reconciler_spec.rb (hunk 8 above)
Comment (LeticiaErrandonea) [reply spec/services/order_details/reconciler_spec.rb:15]: It's now a required param. It can either be `"reconciled"` or `nil`

**Review by giladshanan — APPROVED** [review]
